src/util/collections.ts

```diff
@@ -27,3 +27,24 @@ export function makeMap<K,V>(original: Map<K,V> | null = null, overrides: Map<K,
   }
   return cloned;
 }
+
+export function filter<T>(iter: Iterable<T>, func: (t: T) => bool): T[] {
+  const res: T[] = [];
+  for (let i of iter) {
+    if (func(i)) {
+      res.push(i);
+    }
+  }
+  return res;
+}
+export function map<T, R = T>(iter: Iterable<T>, func: (t: T) => R): R[] {
+  const res: R[] = [];
+  for (let i of iter) {
+      res.push(func(i));
+  }
+  return res;
+}
+
+export function notNull<T>(t: T): bool {
+  return t != null;
+}
```

Comment:
Not a fan of these, as it's trading convenience for the overhead of indirect calls. Would prefer to inline.

Reply:
Yeah I never end up using these, did at first for the convenience but removed most used of iterators after first batch of feedback.

Reply:
Yeah I never ended up using these after removing most of my use of iterations in my first iteration.  I'll remove them.